### PR TITLE
fix(core/data-include): follow element indentation

### PR DESCRIPTION
**Pull request by saschanaz**
A trial to prevent the breakage found in #2650.

### CODE REVIEW

File: tests/spec/core/data-include-spec.js [hunk 1]
```diff
@@ -81,4 +88,38 @@ describe("Core — Data Include", () => {
     expect(h2.textContent).toBe("1. PASS");
     expect(doc.querySelectorAll("*[data-include]").length).toBe(0);
   });
+
+  it("indents multiline text", async () => {
+    const include = `Blue rose is blue
+
+Red rose is red
+
+Rose of the wasteland is violet
```
Comment (marcoscaceres) [inline tests/spec/core/data-include-spec.js]: 🥺😂😭

File: tests/spec/core/data-include-spec.js [hunk 2]
```diff
@@ -81,4 +88,38 @@ describe("Core — Data Include", () => {
...
+      config: {
+        ...makeBasicConfig(),
+        format: "markdown",
+      },
+      body: `
+        ## R
+
+        1. Rose
+
+          <div data-include="${generateDataUrl(
```
Comment (marcoscaceres) [inline tests/spec/core/data-include-spec.js]: This is.... insane... I hope no one actually does this in practice! But very cool that it works.

File: src/core/data-include.js [hunk 3]
```diff
@@ -14,28 +14,76 @@ import { runTransforms } from "./utils.js";
...
+function indentTextWithoutFirstLine(text, indent) {
+  const lines = text.split("\n");
+  const firstLine = lines.shift();
+  return `${firstLine}\n${lines.map(line => indent + line).join("\n")}`;
+}
+
+/**
+ * @param {HTMLElement} el
+ * @param {string} data
+ * @param {boolean} replace
```
Comment (marcoscaceres) [inline src/core/data-include.js]: I kinda like it that TypeScript makes boolean traps less bad. Stylistically tho, I wonder if we should still use a property bag here?

File: tests/spec/core/data-include-spec.js (hunk 2 above)
Comment (saschanaz) [reply tests/spec/core/data-include-spec.js]: Insane, yes 😂

Although this makes more sense in more real-world like example:

```
  <section id="sotd">...</section>

  <section data-include="..."></section>
```

Previously no indentation was applied and thus the replacement text went indentation-less, causing incorrect markdown. I couldn't add this test here though, as `normalizePadding` somehow silences it.

Comment (marcoscaceres) [reply tests/spec/core/data-include-spec.js]: Make sense.

File: src/core/data-include.js [hunk 4]
```diff
@@ -14,28 +14,77 @@ import { runTransforms } from "./utils.js";
...
+}
+
+/**
+ * @param {string} text
+ * @param {string} indent
+ */
+function indentTextWithoutFirstLine(text, indent) {
+  const lines = text.split("\n");
+  const firstLine = lines.shift();
+  return `${firstLine}\n${lines.map(line => indent + line).join("\n")}`;
```
Comment (marcoscaceres) [inline src/core/data-include.js]: clever :)

**Comment by marcoscaceres** [comment]
(not sure if you were ready for review... but looking pretty good)

**Comment by saschanaz** [comment]
I think it's ready, personally would prefer more tests but probably after replacing normalize Padding.

**Comment by marcoscaceres** [comment]
Should we track down some specs that would make use of this to test?

**Comment by saschanaz** [comment]
Basically nearly all specs that we previously observed with data-include need this before we replace normalize Padding 👀

**Comment by saschanaz** [comment]
@marcoscaceres I tried making this a separate PR but I feel this is half-complete without #2715. Some thoughts:

* I added a test here but it's hypothetical rather than a real-world problem.
* Tried merging this in #2715 and immediately found a bug, of which the fix should be both here and #2715.
* #2715 might become too huge to review when merged

What do you think? Does merging the two PR look okay?

**Comment by saschanaz** [comment]
Merged already there, hope it won't be too huge 🙏

**Pull request closed without merging**